src/main/java/seedu/address/logic/commands/sale/EditCommand.java

```diff
@@ -77,36 +80,61 @@ public CommandResult execute(Model model) throws CommandException {
         }
 
-        Sale saleToEdit = lastShownList.get(saleIndex.getZeroBased());
+        List<Sale> editedSales = new ArrayList<>();
+        List<Sale> invalidSales = new ArrayList<>();
+
+        for (Index saleIndex : saleIndexes) {
+            Sale saleToEdit = lastShownList.get(saleIndex.getZeroBased());
+
+            if (personIndex != null) {
+                List<Person> lastShownPeople = model.getSortedPersonList();
+                if (personIndex.getZeroBased() >= lastShownPeople.size()) {
+                    throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+                }
+                Person newBuyer = lastShownPeople.get(personIndex.getZeroBased());
+                editSaleDescriptor.setBuyer(newBuyer);
+            } else {
+                editSaleDescriptor.setBuyer(saleToEdit.getBuyer());
+            }
+
+            Sale editedSale = createEditedSale(saleToEdit, editSaleDescriptor);
 
-        if (personIndex != null) {
-            List<Person> lastShownPeople = model.getSortedPersonList();
-            if (personIndex.getZeroBased() >= lastShownPeople.size()) {
-                throw new CommandException(Messages.MESSAGE_INVALID_PERSON_DISPLAYED_INDEX);
+            if (!model.saleTagsExist(editedSale)) {
+                throw new CommandException(Messages.MESSAGE_SALE_TAGS_NOT_FOUND);
+            }
+
+            if (!saleToEdit.isSameSale(editedSale) && model.hasSale(editedSale)) {
+                invalidSales.add(editedSale);
+            } else {
+                model.setSale(saleToEdit, editedSale);
+                editedSales.add(editedSale);
             }
-            Person newBuyer = lastShownPeople.get(personIndex.getZeroBased());
-            editSaleDescriptor.setBuyerId(newBuyer.getId());
-        } else {
-            editSaleDescriptor.setBuyerId(saleToEdit.getBuyerId());
         }
 
-        Sale editedSale = createEditedSale(saleToEdit, editSaleDescriptor);
+        model.updateFilteredSaleList(PREDICATE_SHOW_ALL_SALES);
+
+        String result = "";
 
-        if (!model.saleTagsExist(editedSale)) {
-            throw new CommandException(Messages.MESSAGE_SALE_TAGS_NOT_FOUND);
+        if (editedSales.size() > 0) {
+            result = MESSAGE_EDIT_SALE_SUCCESS + MassSaleCommandUtil.listAllSales(editedSales);
+        } else {
+            result = MESSAGE_EDIT_SALE_FAILED;
         }
 
-        if (!saleToEdit.isSameSale(editedSale) && model.hasSale(editedSale)) {
-            throw new CommandException(MESSAGE_DUPLICATE_SALE);
+        if (invalidSales.size() > 0) {
+            result += "\n" + MESSAGE_DUPLICATE_SALE + MassSaleCommandUtil.listAllSales(invalidSales);
         }
 
-        model.setSale(saleToEdit, editedSale);
-        model.updateFilteredSaleList(PREDICATE_SHOW_ALL_SALES);
-        return new CommandResult(String.format(MESSAGE_EDIT_SALE_SUCCESS, editedSale));
+        return new CommandResult(result);
```

Comment:
The sale edit command doesn't seem to update sales amounts properly. When I change the buyer of the a certain sale, the sales amounts of the affected contacts aren't updated. Is this the intended behaviour?

Reply:
Good catch, should be fixed in the new commit.